src/models/thread.ts

```diff
@@ -191,7 +191,10 @@ export class Thread extends TypedEventEmitter<EmittedEvents, EventHandlerMap> {
         // There is a risk that the `localTimestamp` approximation will not be accurate
         // when threads are used over federation. That could results in the reply
         // count value drifting away from the value returned by the server
-        if (!this.lastEvent || (isThreadReply && event.localTimestamp > this.lastEvent.localTimestamp)) {
+        if (!this.lastEvent || (isThreadReply
+            && (event.getId() !== this.lastEvent.getId())
```

Comment:
would be good to get a comment here to say why we are checking the IDs (it implies there is code somewhere that is dealing with the `lastEvent`'s count, but not sure)